include/brickengine/input.hpp

```diff
@@ -6,30 +6,55 @@
 #include <algorithm>
 #include <tuple>
 #include <utility>
+#include <map>
 
 #include "SDL2/SDL.h"
 #include "brickengine/input_keycode.hpp"
 
 template<typename T>
 class BrickInput {
 public:
-    BrickInput(){
+    BrickInput() {
         generateInputs();
+        if(SDL_GameControllerAddMappingsFromFile("assets/controller/gamecontrollerdb.txt") == -1) {
+            std::cout << "error" << std::endl;
+        }
+        // Checking for joysticks
+        for (int i = 0; i < SDL_NumJoysticks(); ++i) {
+            auto controller = SDL_GameControllerOpen(i);
+            auto id = SDL_JoystickInstanceID(SDL_GameControllerGetJoystick(controller));
+            if(controller) {
+                std::cout << "Found controller: " << SDL_GameControllerNameForIndex(i) << std::endl;
+                controllers[id] = controller;
+            } else {
+                std::cout << "Couldn't open joystick: " << i << ", skipping...";
+            }
+        }
     }
-
+    
     static BrickInput<T>& getInstance() {
         static BrickInput<T> INSTANCE;
         return INSTANCE;
     }
 
-    void setInputMapping(std::unordered_map<int, std::unordered_map<InputKeyCode, T>>& game_input,
-                         std::unordered_map<T, double>& time_to_wait_mapping) {
+    void setInputMapping(std::map<int, std::unordered_map<InputKeyCode, T>>& game_input,
+                         std::unordered_map<T, double>& time_to_wait_mapping,
+                         std::unordered_map<InputKeyCode, signed int> value_map) {
+        axis_map = value_map;
         input_mapping = game_input;
+        int count = 0;
         for (auto& [player_id, mapping] : input_mapping) {
+            // Bind players to controllers
+            //if(count < controllers.size()) {
+            //    std::cout << "Binding controller to player: " << player_id << std::endl;
+            //    player_controller_mapping[player_id] = SDL_JoystickInstanceID(SDL_GameControllerGetJoystick(controllers.at(count)));
+            //}
+
```

Comment:
Is this going to be used in the future or can it be removed?

Reply:
This is not necessary anymore since JOYDEVICEADDED is called when the game starts.